code/frameworks/angular/src/client/public-types.ts

```diff
@@ -21,27 +23,54 @@ export type { AngularRenderer };
  *
  * @see [Default export](https://storybook.js.org/docs/formats/component-story-format/#default-export)
  */
-export type Meta<TArgs = Args> = ComponentAnnotations<AngularRenderer, TransformEventType<TArgs>>;
+export type Meta<TArgs = Args> = ComponentAnnotations<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 /**
  * Story function that represents a CSFv2 component example.
  *
  * @see [Named Story exports](https://storybook.js.org/docs/formats/component-story-format/#named-story-exports)
  */
-export type StoryFn<TArgs = Args> = AnnotatedStoryFn<AngularRenderer, TransformEventType<TArgs>>;
+export type StoryFn<TArgs = Args> = AnnotatedStoryFn<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 /**
  * Story object that represents a CSFv3 component example.
  *
  * @see [Named Story exports](https://storybook.js.org/docs/formats/component-story-format/#named-story-exports)
  */
-export type StoryObj<TArgs = Args> = StoryAnnotations<AngularRenderer, TransformEventType<TArgs>>;
+export type StoryObj<TArgs = Args> = StoryAnnotations<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 export type Decorator<TArgs = StrictArgs> = DecoratorFunction<AngularRenderer, TArgs>;
 export type Loader<TArgs = StrictArgs> = LoaderFunction<AngularRenderer, TArgs>;
 export type StoryContext<TArgs = StrictArgs> = GenericStoryContext<AngularRenderer, TArgs>;
 export type Preview = ProjectAnnotations<AngularRenderer>;
 
+/**
+ * Utility type that transforms InputSignal and EventEmitter types
+ */
+type TransformComponentType<T> = TransformInputSignalType<TransformEventType<T>>
+
+// @ts-ignore Angular < 17.2 doesn't export InputSignal
+type AngularInputSignal<T> = AngularCore.InputSignal<T>
+// @ts-ignore Angular < 17.2 doesn't export InputSignalWithTransform
+type AngularInputSignalWithTransform<T, U> = AngularCore.InputSignalWithTransform<T, U>
+
+type AngularHasSignal = typeof AngularCore extends { input: infer U } ? true : false;
```

Comment:
That is possible? `import * as X` and X is basically a namespace/interface for everything exported inside of it? 🤯 
Should there be a small comment that gives a hint about what's happening here? I leave it up to you